cg/meta/upload/fohm/fohm.py

```diff
@@ -0,0 +1,199 @@
+import logging
+import shutil
+from pathlib import Path
+from typing import List, Optional
+import datetime as dt
+import pandas as pd
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.apps.lims import LimsAPI
+from cg.models.cg_config import CGConfig
+from cg.models.email import EmailInfo
+from cg.store import Store, models
+from cg.utils.email import send_mail
+from housekeeper.store.models import Version
+
+LOG = logging.getLogger(__name__)
+
+
+class FOHMUploadAPI:
+    def __init__(self, config: CGConfig):
+        self.config: CGConfig = config
+        self.housekeeper_api: HousekeeperAPI = config.housekeeper_api
+        self.lims_api: LimsAPI = config.lims_api
+        self.status_db: Store = config.status_db
+
+        self._current_datestr = None
+        self._daily_bundle_path = None
+        self._daily_rawdata_path = None
+        self._daily_report_path = None
+        self._cases_to_aggregate: List[str] = []
+        self._daily_reports_list: List[Path] = []
+        self._daily_pangolin_list: List[Path] = []
+        self._reports_dataframe = None
+        self._pangolin_dataframe = None
+        self._aggregation_dataframe = None
+
+    @property
+    def current_datestr(self) -> str:
+        if not self._current_datestr:
+            self._current_datestr = str(dt.date.today())
+        return self._current_datestr
+
+    @property
+    def daily_bundle_path(self) -> Path:
+        if not self._daily_bundle_path:
+            self._daily_bundle_path = (
+                Path(self.config.mutant.root).parent / "fohm" / self.current_datestr
+            )
+        return self._daily_bundle_path
+
+    @property
+    def daily_rawdata_path(self) -> Path:
+        if not self._daily_rawdata_path:
+            self._daily_rawdata_path = self.daily_bundle_path / "rawdata"
+        return self._daily_rawdata_path
+
+    @property
+    def daily_report_path(self) -> Path:
+        if not self._daily_report_path:
+            self._daily_report_path = self.daily_bundle_path / "reports"
+        return self._daily_report_path
+
+    @property
+    def reports_dataframe(self) -> pd.DataFrame:
+        """Dataframe with all komplettering rows from multiple cases"""
+        if not isinstance(self._reports_dataframe, pd.DataFrame):
+            self._reports_dataframe = self.create_joined_dataframe(
+                self.daily_reports_list
+            ).sort_values(by=["provnummer"])
+        return self._reports_dataframe
+
+    @property
+    def pangolin_dataframe(self) -> pd.DataFrame:
```

Comment:
Could not `reports_dataframe` and `pangolin_dataframe` be made into one if you send in the dataframe and a tag?